src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java

```diff
@@ -0,0 +1,123 @@
+                    CaseData caseData = submitEvent.getCaseData();
+                    setJudgeName(caseData, existingJudgeCode, requiredJudgeCode);
+
+                    CCDRequest returnedRequest = ccdClient.startEventForCase(authToken, caseTypeId,
+                            refDataFixesDetails.getJurisdiction(), String.valueOf(submitEvent.getCaseId()));
+                    ccdClient.submitEventForCase(authToken, caseData, caseTypeId,
+                            refDataFixesDetails.getJurisdiction(), returnedRequest, String.valueOf(submitEvent.getCaseId()));
+                }
+                log.info(String.format(
+                        "Existing Judge's code in all cases from %s to %s is " +
+                                "updated to required judge's code", dateFrom, dateTo));
+            }
+            return refDataFixesData;
+        } catch (Exception ex) {
+            log.error(MESSAGE + refDataFixesDetails.getCaseId(), ex);
+            throw new CaseRetrievalException(MESSAGE + refDataFixesDetails.getCaseId() + ex.getMessage());
+        }
+    }
+
+    private List<String> getDateRangeForSearch(RefDataFixesDetails refDataFixesDetails) {
+        var refDataFixesData = refDataFixesDetails.getCaseData();
+        boolean isRangeHearingDateType = refDataFixesData.getHearingDateType().equals(RANGE_HEARING_DATE_TYPE);
+        String dateFrom;
+        String dateTo;
+        if (!isRangeHearingDateType) {
+            dateFrom = LocalDate.parse(refDataFixesData.getDate(), OLD_DATE_TIME_PATTERN2)
+                    .atStartOfDay().format(OLD_DATE_TIME_PATTERN);
+            dateTo = LocalDate.parse(refDataFixesData.getDate(), OLD_DATE_TIME_PATTERN2)
+                    .atStartOfDay().plusDays(1).minusSeconds(1).format(OLD_DATE_TIME_PATTERN);
+        } else {
+            dateFrom = LocalDate.parse(refDataFixesData.getDateFrom(), OLD_DATE_TIME_PATTERN2)
+                    .atStartOfDay().format(OLD_DATE_TIME_PATTERN);
+            dateTo = LocalDate.parse(refDataFixesData.getDateTo(), OLD_DATE_TIME_PATTERN2)
+                    .atStartOfDay().plusDays(1).minusSeconds(1).format(OLD_DATE_TIME_PATTERN);
+        }
+        return new ArrayList<>(List.of(dateFrom, dateTo));
+    }
+
+    private void setJudgeName(CaseData caseData, String existingJudgeCode, String requiredJudgeCode) {
+        for (HearingTypeItem hearingTypeItem : caseData.getHearingCollection()) {
+            HearingType hearingType = hearingTypeItem.getValue();
+            if (!Strings.isNullOrEmpty(hearingType.getJudge()) && hearingType.getJudge().equals(existingJudgeCode)) {
+                hearingType.setJudge(requiredJudgeCode);
+                log.info(String.format("Judge's code in Case with ethosReference %s is updated", caseData.getEthosCaseReference()));
+            }
+        }
+    }
+
+//    public RefDataFixesData insertClaimServedDate(RefDataFixesDetails refDataFixesDetails, String authToken) {
```

Comment:
This commented out code has nothing to do with the fix you're doing

Reply:
removed